pkg/wallet/wallet.go

```diff
@@ -175,41 +178,69 @@ func Generate(nickname string, password string) (*Wallet, error) {
 
 	addr := blake3.Sum256(publicKey)
 
-	var salt [16]byte
+	return CreateWalletFromKeys(nickname, privateKey, publicKey, addr, password)
+
+}
 
-	_, err = rand.Read(salt[:])
+// Delete removes wallet from file system
+func Delete(nickname string) (err error) {
+	err = os.Remove(Filename(nickname))
 	if err != nil {
-		return nil, fmt.Errorf("generating random salt: %w", err)
+		return fmt.Errorf("deleting wallet '%s': %w", Filename(nickname), err)
 	}
 
-	var nonce [12]byte
+	return nil
+}
 
-	_, err = rand.Read(nonce[:])
+// Filename returns the wallet Filename based on the given nickname.
+func Filename(nickname string) string {
+	return fmt.Sprintf("wallet_%s.json", nickname)
+}
+
+func Import(nickname string, privateKeyB58V string, password string) (*Wallet, error) {
+	privateKeyBytes, _, err := base58.CheckDecode(privateKeyB58V[1:])
 	if err != nil {
-		return nil, fmt.Errorf("generating random nonce: %w", err)
+		return nil, fmt.Errorf("encoding private key B58: %w", err)
 	}
 
-	wallet, err := New(nickname, addr, privateKey, publicKey, salt, nonce)
+	wallets, err := LoadAll()
 	if err != nil {
-		return nil, fmt.Errorf("instantiating a new wallet: %w", err)
+		return nil, fmt.Errorf("error loading wallets %w", err)
 	}
 
-	err = wallet.Protect(password)
-	if err != nil {
-		return nil, fmt.Errorf("protecting the new wallet: %w", err)
+	// The ed25519 seed is in fact what we call a private key in cryptography...
+	privateKey := ed25519.NewKeyFromSeed(privateKeyBytes)
+
+	pubKeyBytes := reflect.ValueOf(privateKey.Public()).Bytes() // force conversion to byte array
+
+	addr := blake3.Sum256(pubKeyBytes)
+	version := byte(0)
+	address := "A" + base58.CheckEncode(addr[:], version)
+
+	if slices.IndexFunc(
+		wallets,
+		func(wallet Wallet) bool { return wallet.Address == address },
+	) != -1 {
+		return nil, fmt.Errorf("Error: Address already exists.")
```

Comment:
The pattern for error is stric: `<what you are doing>: <the error>`
```suggestion
		return nil, fmt.Errorf("importing new wallet: duplicate wallet with different name (but same keys).")
```